cypress/support/commands.js

```diff
@@ -0,0 +1,142 @@
+import 'cypress-axe';
+import { faker } from '@faker-js/faker';
+
+// Comando personalizado para simular falha de contraste de cores
+Cypress.Commands.add('simulateContrastFailure', () => {
+  cy.get('body').then(($body) => {
+    $body.find('h1, h2, h3, p, span').each((index, el) => {
+      cy.wrap(el).invoke('css', 'color', '#dddddd'); // Texto claro
+      cy.wrap(el).invoke('css', 'background-color', '#ffffff'); // Fundo claro
+    });
+  });
+});
+
+// Comando personalizado para simular falta de alt em uma imagem
+Cypress.Commands.add('simulateMissingAlt', () => {
+  cy.get('body').then(($body) => {
+    $body.append('<img src="https://via.placeholder.com/150" />'); // Imagem sem alt
+  });
+});
+
+// Comando personalizado para simular um botão sem aria-label
+Cypress.Commands.add('simulateMissingAriaLabel', () => {
+  cy.get('body').then(($body) => {
+    $body.append('<button></button>'); // Botão sem aria-label ou texto visível
+  });
+});
+
+
+Cypress.Commands.add('validateslinksInTheFooter', () => {
+  cy.contains('p', 'Copyright 2024 - Talking About Testing').should('be.visible');
+  const stringsToValidate = [
+    'Hotmart',
+    'Udemy',
+    'Blog',
+    'YouTube'
+  ];
+  stringsToValidate.forEach((link) => {
+    cy.contains(link).should('exist');
+  });
+});
+
+//Valida o texto de saudação não informando um nome no campo de localização de cliente
+Cypress.Commands.add('validatesGreetingNoGivenName', () => {
+
+  cy.get('h2').within(() => {
+    cy.contains('Hi').should('exist');
+    cy.contains('there').should('exist');
+    cy.contains('!').should('exist');
+  });
+});
+
+//Valida o texto de saudação informando um nome no campo de localização de cliente
+Cypress.Commands.add('validatesGreetingGivenName', () => {
+
+  const clientName = faker.person.firstName();
+  cy.get('#name').type(clientName);
+
+  cy.get('h2').within(() => {
```

Comment:
E aqui podia ser `cy.contains('h2', 'Hi Joe!').should('be.visible')`.

Reply:
Desde que o nome `Joe` fosse digitado.

Reply:
Tem razão deixei o código desta forma e retirei a função personalizada